### PR TITLE
Defer start of pspdelete user controller

### PR DESCRIPTION
**Pull request by ibuildthecloud**
Only start pspdelete user controller if DefaultPodSecurityPolicyTemplateName is set on the v3.Cluster and its RKE2 managed cluster.  The pspdelete controller is used specifically for RKE2 to PSP with "psp.rke2.io/global-unrestricted" and "psp.rke2.io/global-restricted" annotations that are created by default for RKE2.

### CODE REVIEW

File: pkg/controllers/managementuser/pspdelete/pspdelete.go [hunk 1]
```diff
@@ -22,7 +23,36 @@ type handler struct {
...
+	})
+
+	clusters := userContext.Management.Management.Clusters("")
+	clusterCache := userContext.Management.Wrangler.Provisioning.Cluster().Cache()
+	clusters.AddHandler(ctx, "pspdelete-deferred", func(key string, obj *v3.Cluster) (runtime.Object, error) {
+		if obj != nil &&
+			obj.Name == userContext.ClusterName &&
+			obj.Spec.DefaultPodSecurityPolicyTemplateName != "" {
+
+		}
```
Comment (rmweir) [inline pkg/controllers/managementuser/pspdelete/pspdelete.go:39]: empty block here, guessing you meant to return

File: pkg/controllers/managementuser/pspdelete/pspdelete.go [hunk 2]
```diff
@@ -22,7 +23,36 @@ type handler struct {
...
+		clusters, err := clusterCache.GetByIndex(cluster.ByCluster, userContext.ClusterName)
```
Comment (rmweir) [inline pkg/controllers/managementuser/pspdelete/pspdelete.go:40]: why use indexer instead of just getting it off the cache regularly?

**Review by rmweir — CHANGES_REQUESTED** [review]

File: pkg/controllers/managementuser/pspdelete/pspdelete.go (hunk 2 above)
Comment (rmweir) [reply pkg/controllers/managementuser/pspdelete/pspdelete.go:40]: Answer offline, resolved.

File: pkg/controllers/managementuser/pspdelete/pspdelete.go [hunk 3]
```diff
@@ -22,7 +23,37 @@ type handler struct {
...
+			return obj, nil
+		}
+
+		clusters, err := clusterCache.GetByIndex(cluster.ByCluster, obj.Name)
+		if err != nil || len(clusters) != 1 {
+			return obj, err
+		}
+
+		clusterSpec := clusters[0].Spec
+		if clusterSpec.DefaultPodSecurityPolicyTemplateName != "" && clusters[0].Spec.RKEConfig != nil {
```
Comment (rmweir) [inline pkg/controllers/managementuser/pspdelete/pspdelete.go:47]: nit:
```suggestion
		if clusterSpec.DefaultPodSecurityPolicyTemplateName != "" && clustersSpec.RKEConfig != nil {
```

**Review by rmweir — DISMISSED** [review]
one nit, then LGTM

**Review by deniseschannon — APPROVED** [review]

**Review by deniseschannon — APPROVED** [review]

**Pull request merged**